Source/SwiftLintBuiltInRules/Rules/Performance/EmptyCountRule.swift

```diff
@@ -59,6 +97,67 @@ private extension EmptyCountRule {
             }
         }
     }
+
+    final class Rewriter: ViolationsSyntaxRewriter {
+        private let operators: Set = ["==", "!=", ">", ">=", "<", "<="]
+        private let configuration: EmptyCountConfiguration
+
+        init(configuration: EmptyCountConfiguration,
+             locationConverter: SourceLocationConverter,
+             disabledRegions: [SourceRange]) {
+            self.configuration = configuration
+            super.init(locationConverter: locationConverter, disabledRegions: disabledRegions)
+        }
+
+        override func visit(_ node: SequenceExprSyntax) -> ExprSyntax {
+            guard let folded = try? OperatorTable.standardOperators.foldSingle(node) else { return super.visit(node) }
+
+            if let infix = folded.as(InfixOperatorExprSyntax.self) {
+                return visit(infix)
+            } else {
+                return super.visit(folded)
+            }
+        }
+
+        override func visit(_ node: InfixOperatorExprSyntax) -> ExprSyntax {
+            guard let operatorNode = node.operator.as(BinaryOperatorExprSyntax.self),
+                  let binaryOperator = operatorNode.operator.binaryOperator,
+                  operators.contains(binaryOperator) else {
+                return super.visit(node)
+            }
+
+            if let (count, position) = node.countNodeAndPosition(onlyAfterDot: configuration.onlyAfterDot) {
+                let newNode: ExprSyntax? = if let count = count.as(MemberAccessExprSyntax.self) {
+                    count.with(\.declName.baseName, "isEmpty").trimmed.as(ExprSyntax.self)
+                } else if let count = count.as(DeclReferenceExprSyntax.self) {
+                    count.with(\.baseName, "isEmpty").trimmed.as(ExprSyntax.self)
+                } else {
+                    nil
+                }
+
+                if let newNode {
+                    correctionPositions.append(position)
+                    if ["!=", "<", ">"].contains(binaryOperator) {
```

Comment:
`if` expression? 😊

Reply:
Fixed at 239829d

I moved `if` expressions around and felt like this looks clean but let me know your thoughts. (The outmost `if` was left as statements since it had few lines of non-expression and was a bit awkward to turn it into an `if` expression.)